js-sandbox/src/call_args.rs

```diff
@@ -3,11 +3,33 @@
 use crate::AnyError;
 use serde::Serialize;
 
+mod sealed {
+	use super::*;
+
+	pub trait Sealed {}
+	impl Sealed for () {}
+
+	macro_rules! impl_sealed {
+		($($param:ident),+) => {
+			#[allow(non_snake_case)] // use generic params as variable names
+			impl<$($param),+> Sealed for ($($param),+,)
+				where $($param : Serialize),+
+			{ }
+		}
+	}
+
+	impl_sealed!(P0);
+	impl_sealed!(P0, P1);
+	impl_sealed!(P0, P1, P2);
+	impl_sealed!(P0, P1, P2, P3);
+	impl_sealed!(P0, P1, P2, P3, P4);
+}
```

Comment:
I merged the macros for readability

I always preferred the supertrait seal for non-marker traits vs the method argument sealed struct, since it lets downstream code call implemented methods, but still prevents additional downstream implementations